File: webapi/Models/Request/Ask.cs
// Copyright (c) Microsoft. All rights reserved.

using System.ComponentModel.DataAnnotations;
using CopilotChat.WebApi.Options;

namespace CopilotChat.WebApi.Models.Request;

public class Ask
{
    [Required, NotEmptyOrWhitespace]
    public string Input { get; set; } = string.Empty;

    public IEnumerable<KeyValuePair<string, string>> Variables { get; set; } = Enumerable.Empty<KeyValuePair<string, string>>();
}

File: webapp/src/libs/semantic-kernel/model/Ask.ts
// Copyright (c) Microsoft. All rights reserved.

export interface IAsk {
    input: string;
    variables?: IAskVariables[];
}

export interface IAskVariables {
    key: string;
    value: string;
}

export interface IAskSearch {
    specializationKey: string;
    search: string;
}
